syntax = "proto3";
package cronus;

message Properties {
  repeated Property property = 1;
}

message Property {
  string name = 1;
  string type = 2;
  string value = 3;
}

message Module {
  string name = 1;
  string module = 2;
  string klass = 3;
  Properties properties = 4;
}

message GeneratorInput {
  Module config = 1;
}

message AtomInput {
  string name = 1;
  string repo = 2;
  string glob = 3;
}

message Input {
  GeneratorInput generator = 1;
  AtomInput atom = 2;
}

message Sampler {
  int32 ndatums = 1;
  int32 nchunks = 2;
}

message Configuration {
  string name = 1;
  string uuid = 2;
  repeated Module tools = 3;
  repeated Module algos = 4;
  uint64 max_malloc_size_bytes = 5;
  Sampler sampler = 6;
  Input input = 7;
}

